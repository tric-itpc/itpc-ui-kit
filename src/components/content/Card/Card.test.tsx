import React from "react"

import { render, screen } from "@testing-library/react"

import { Card } from "./index"

describe("Компонент Card", () => {
  test("Снимок Card без заголовка и детей", () => {
    const { asFragment } = render(<Card title="Test Card" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("Снимок Card с заголовком и детьми", () => {
    const { asFragment } = render(
      <Card className="custom-class" title="Test Card">
        <div>Content</div>
      </Card>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Снимок Card с рамкой", () => {
    const { asFragment } = render(<Card title="Test Card" isBordered />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("Рендер Card без заголовка и детей", () => {
    const { container } = render(<Card />)
    const card = container.querySelector(".itpc-card")

    expect(card).toHaveClass("itpc-card")

    const body = card?.querySelector(".itpc-card__body")
    expect(body).toBeInTheDocument()
    expect(body).toBeEmptyDOMElement()
  })

  test("Рендер Card с заголовком", () => {
    render(<Card title="Заголовок карточки" />)
    const title = screen.getByText("Заголовок карточки")
    expect(title).toHaveClass("itpc-card__title")
  })

  test("Рендер Card с детьми", () => {
    render(
      <Card>
        <p>Содержимое карточки</p>
      </Card>
    )
    const bodyContent = screen.getByText("Содержимое карточки")
    expect(bodyContent).toBeInTheDocument()
  })

  test("Присутствие рамки при isBordered=true", () => {
    const { container } = render(<Card isBordered />)
    const card = container.querySelector(".itpc-card")

    expect(card).toBeInTheDocument()

    expect(card).toHaveClass("itpc-card", "itpc-card_bordered")
  })

  test("Проверка передачи дополнительных атрибутов в bodyAttr", () => {
    render(
      <Card bodyAttr={{ "data-testid": "card-body" }}>
        <p>Контент</p>
      </Card>
    )
    const cardBody = screen.getByTestId("card-body")
    expect(cardBody).toHaveClass("itpc-card__body")
  })

  test("Проверка передачи дополнительных атрибутов в titleAttr", () => {
    render(
      <Card
        titleAttr={{
          "aria-label": "Тестовый заголовок",
          "data-testid": "card-title",
        }}
        title="Заголовок"
      />
    )
    const cardTitle = screen.getByTestId("card-title")
    expect(cardTitle).toHaveAttribute("aria-label", "Тестовый заголовок")
    expect(cardTitle).toHaveClass("itpc-card__title")
  })

  test("Присваивание дополнительного класса через className", () => {
    const { container } = render(<Card className="custom-class" />)
    const card = container.querySelector(".itpc-card")

    expect(card).toBeInTheDocument()

    expect(card).toHaveClass("itpc-card", "custom-class")
  })

  test("Рендер Card со всеми пропсами", () => {
    const { container } = render(
      <Card
        bodyAttr={{ "data-testid": "card-body" }}
        className="custom-class"
        title="Заголовок карточки"
        titleAttr={{ "data-testid": "card-title" }}
        isBordered
      >
        <p>Содержимое карточки</p>
      </Card>
    )

    const card = container.querySelector(".itpc-card")
    expect(card).toBeInTheDocument()
    expect(card).toHaveClass("itpc-card", "itpc-card_bordered", "custom-class")

    const title = container.querySelector("[data-testid='card-title']")
    expect(title).toBeInTheDocument()
    expect(title).toHaveClass("itpc-card__title")
    expect(title).toHaveTextContent("Заголовок карточки")

    const body = container.querySelector("[data-testid='card-body']")
    expect(body).toBeInTheDocument()
    expect(body).toHaveClass("itpc-card__body")
    expect(body).toHaveTextContent("Содержимое карточки")
  })
})
