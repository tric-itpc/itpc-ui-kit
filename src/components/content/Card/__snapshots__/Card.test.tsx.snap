// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Card Снимок Card без заголовка и детей 1`] = `
<DocumentFragment>
  <div
    class="itpc-card"
  >
    <p
      class="itpc-card__title"
    >
      Test Card
    </p>
    <div
      class="itpc-card__body"
    />
  </div>
</DocumentFragment>
`;

exports[`Компонент Card Снимок Card с заголовком и детьми 1`] = `
<DocumentFragment>
  <div
    class="itpc-card custom-class"
  >
    <p
      class="itpc-card__title"
    >
      Test Card
    </p>
    <div
      class="itpc-card__body"
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Card Снимок Card с рамкой 1`] = `
<DocumentFragment>
  <div
    class="itpc-card itpc-card_bordered"
  >
    <p
      class="itpc-card__title"
    >
      Test Card
    </p>
    <div
      class="itpc-card__body"
    />
  </div>
</DocumentFragment>
`;
